Fenwick_Tree/Fenwick_Tree.dart

```diff
@@ -0,0 +1,164 @@
+/**
+ *    Fenwick Tree
+ * -------------------------
+ * Binary Indexed tree or Fenwick tree is used to solve
+ * range queries optimally. It has O(n) space complexity,
+ * O(logn) time complexity to update, O(nlog(n)) time complexity to build
+ * and O(logn) time complexity to find sum in a given range.
+ * 
+ * References - Geeksforgeeks
+ * 
+ */
+
+
+// Importing required libraries
```

Comment:
please do the changes here as i asked you to do in other PR

Reply:
@asha15 I wrote in my own words in every PR I made. I only used GFG to learn and understand the algorithm. I hope you understand. But if you still think I should change, I will do it, please let me know.